File: util/src/entities/Sticker.ts
import { Column, Entity, JoinColumn, ManyToOne, RelationId } from "typeorm";
import { User } from "./User";
import { BaseClass } from "./BaseClass";
import { Guild } from "./Guild";

export enum StickerType {
	STANDARD = 1,
	GUILD = 2,
}

export enum StickerFormatType {
	PNG = 1,
	APNG = 2,
	LOTTIE = 3,
}

@Entity("stickers")
export class Sticker extends BaseClass {
	@Column()
	name: string;

	@Column({ nullable: true })
	description?: string;

	@Column()
	tags: string;

	@Column()
	@RelationId((sticker: Sticker) => sticker.pack)
	pack_id: string;

	@JoinColumn({ name: "pack_id" })
	@ManyToOne(() => require("./StickerPack").StickerPack, {
		onDelete: "CASCADE",
	})
	pack: import("./StickerPack").StickerPack;

	@Column({ nullable: true })
	guild_id?: string;

	@JoinColumn({ name: "guild_id" })
	@ManyToOne(() => Guild, {
		onDelete: "CASCADE",
	})
	guild?: Guild;

	@Column({ nullable: true })
	user_id?: string;

	@JoinColumn({ name: "user_id" })
	@ManyToOne(() => User, {
		onDelete: "CASCADE",
	})
	user?: User;

	@Column({ type: "int" })
	type: StickerType;

	@Column({ type: "int" })
	format_type: StickerFormatType;
}

File: extra/admin-api/Models/Spacebar.Models.Db/Models/Sticker.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Spacebar.Models.Db.Models;

[Table("stickers")]
public partial class Sticker
{
    [Key]
    [Column("id")]
    public long Id { get; set; }

    [Column("name", TypeName = "character varying")]
    public string Name { get; set; } = null!;

    [Column("description", TypeName = "character varying")]
    public string? Description { get; set; }

    [Column("available")]
    public bool? Available { get; set; }

    [Column("tags", TypeName = "character varying")]
    public string? Tags { get; set; }

    [Column("pack_id")]
    public long? PackId { get; set; }

    [Column("guild_id")]
    public long? GuildId { get; set; }

    [Column("user_id")]
    public long? UserId { get; set; }

    [Column("type")]
    public int Type { get; set; }

    [Column("format_type")]
    public int FormatType { get; set; }

    [ForeignKey("GuildId")]
    [InverseProperty("Stickers")]
    public virtual Guild? Guild { get; set; }

    [ForeignKey("PackId")]
    [InverseProperty("Stickers")]
    public virtual StickerPack? Pack { get; set; }

    [InverseProperty("CoverStickerId1Navigation")]
    public virtual ICollection<StickerPack> StickerPacks { get; set; } = new List<StickerPack>();

    [ForeignKey("UserId")]
    [InverseProperty("Stickers")]
    public virtual User? User { get; set; }

    [ForeignKey("StickersId")]
    [InverseProperty("Stickers")]
    public virtual ICollection<Message> Messages { get; set; } = new List<Message>();
}
